### PR TITLE
Feature/cli

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

- WP CLI update, adding parent command names, adding examples, fixing resources, etc.
- Updating internal links to use helper for paths so it can be used in tests and on real project
- Fixing tests

# Screenshots / Videos
Screenshot 2022-07-05 at 09 23 24 (2)
Screenshot 2022-07-05 at 09 23 24 (3)
Screenshot 2022-07-05 at 09 24 46
Screenshot 2022-07-05 at 10 18 06

### CODE REVIEW

File: src/Blocks/AbstractBlocksCli.php [hunk 1]
```diff
@@ -19,152 +20,205 @@
...
+						$this->getShortenCliPathOutput($destination)
+					)
+				);
+			}
+		}
 	}
 
 	/**
-	 * Move block/component from frontend libs to project.
+	 * Determin if item has dependencies and output helper commands.
```
Comment (dingo-d) [inline src/Blocks/AbstractBlocksCli.php:190]:
```suggestion
	 * Determine if the item has dependencies and output helper commands.
```

File: src/Blocks/UseComponentCli.php [hunk 2]
```diff
@@ -71,7 +71,7 @@ public function getDoc(): array
 				[
 					'type' => 'assoc',
 					'name' => 'name',
-					'description' => 'Specify component name.',
+					'description' => 'Specify component name. You can specify multiple components by comma separator.',
```
Comment (dingo-d) [inline src/Blocks/UseComponentCli.php:74]:
```suggestion
					'description' => 'Specify component name. You can specify multiple components (separate them with comma).',
```

File: src/Blocks/UseGlobalAssetsCli.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	 * @return array<string, array<int, array<string, bool|string>>|string>
+	 */
+	public function getDoc(): array
+	{
+		return [
+			'shortdesc' => 'Copy global assets from our library to your project.',
+			'longdesc' => $this->prepareLongDesc("
+				## USAGE
+
+				Used to copy pre-created global assets from our library to your project. After copying you can modify it in any way you see fit.
```
Comment (dingo-d) [inline src/Blocks/UseGlobalAssetsCli.php:54]:
```suggestion
				Used to copy pre-created global assets from our library to your project. After copying you can modify them in any way you see fit.
```

File: src/Blocks/UseManifestCli.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	 * Get WPCLI command doc
+	 *
+	 * @return array<string, array<int, array<string, bool|string>>|string>
+	 */
+	public function getDoc(): array
+	{
+		return [
+			'shortdesc' => 'Copy global settings manifest.json file from our library to your project.',
+			'longdesc' => $this->prepareLongDesc("
+				This file is a block editor main setting file where you can find color, option and much more.
```
Comment (dingo-d) [inline src/Blocks/UseManifestCli.php:52]:
```suggestion
				This file is a block editor's main setting file where you can find the color, options, and much more.
```

File: src/Blocks/UseManifestCli.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+				## RESOURCES
+
+				Manifest.json file will be created from this file:
+				https://github.com/infinum/eightshift-frontend-libs/tree/develop/blocks/init/src/Blocks/manifest.json
+			"),
+		];
+	}
+
+	/* @phpstan-ignore-next-line */
+	public function __invoke(array $args, array $assocArgs) // phpcs:ignore
```
Comment (dingo-d) [inline src/Blocks/UseManifestCli.php:69]:
```suggestion
	public function __invoke(array $args, array $assocArgs)
```

File: src/Blocks/UseVariationCli.php [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+				## RESOURCES
+
+				All our variations can be found here:
+				https://github.com/infinum/eightshift-frontend-libs/tree/develop/blocks/init/src/Blocks/variations
+			"),
+		];
+	}
+
+	/* @phpstan-ignore-next-line */
+	public function __invoke(array $args, array $assocArgs) // phpcs:ignore
```
Comment (dingo-d) [inline src/Blocks/UseVariationCli.php:90]:
```suggestion
	public function __invoke(array $args, array $assocArgs)
```

File: src/Blocks/UseWrapperCli.php [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+				## RESOURCES
+
+				Our wrapper can be found here:
+				https://github.com/infinum/eightshift-frontend-libs/tree/develop/blocks/init/src/Blocks/wrapper
+			"),
+		];
+	}
+
+	/* @phpstan-ignore-next-line */
+	public function __invoke(array $args, array $assocArgs) // phpcs:ignore
```
Comment (dingo-d) [inline src/Blocks/UseWrapperCli.php:66]:
```suggestion
	public function __invoke(array $args, array $assocArgs)
```

File: src/CiExclude/CiExcludeCli.php [hunk 8]
```diff
@@ -110,13 +107,19 @@ public function getDoc(): array
...
-		$this->getExampleTemplate(__DIR__, 'ci-exclude.txt')
-			->renameProjectName($assocArgs)
-			->renameProjectType($assocArgs)
-			->outputWrite($root, 'ci-exclude.txt', $assocArgs);
+		$this->getExampleTemplate(__DIR__, $this->getClassShortName())
+			->searchReplaceString('<?php $output = \'', '')
+			->searchReplaceString('\';', '')
+			->searchReplaceString($this->getArgTemplate('project_name'), $projectName)
+			->searchReplaceString($this->getArgTemplate('project_type'), $projectType)
+			->outputWrite($path, 'ci-exclude.txt', $assocArgs);
```
Comment (dingo-d) [inline src/CiExclude/CiExcludeCli.php:123]: We can delete this as this is no longer used anywhere.

File: src/Cli/README.md [hunk 9]
```diff
@@ -3,19 +3,7 @@
...
-```wp eval-file bin/cli.php develop_show_all --skip-wordpress```
-
-### Run all commands:
-
-```wp eval-file bin/cli.php develop_show_all --skip-wordpress```
-
-### Reset output:
-
-```wp eval-file bin/cli.php develop_reset --skip-wordpress```
+You can run this commands by mocking them in the tests.
```
Comment (dingo-d) [inline src/Cli/README.md:6]: I think you can just remove this readme, don't think it serves a purpose 🤷🏼‍♂️

File: src/Build/BuildCli.php [hunk 10]
```diff
@@ -109,13 +106,15 @@ public function getDoc(): array
...
 		// Get Props.
-		$root = $assocArgs['root'] ?? static::OUTPUT_DIR;
+		$path = $this->getArg($assocArgs, 'path');
 
 		// Read the template contents, and replace the placeholders with provided variables.
 		$this->getExampleTemplate(__DIR__, $this->getClassShortName())
 			->renameProjectName($assocArgs)
 			->renameProjectType($assocArgs)
-			->outputWrite($root . 'bin', 'build.sh', $assocArgs);
+			->outputWrite($path, 'build.sh', $assocArgs);
```
Comment (dingo-d) [inline src/Build/BuildCli.php:118]: This file can be deleted, we're not using it anywhere.

File: src/Cli/ParentGroups/CliBlocks.php [hunk 11]
```diff
@@ -18,13 +18,13 @@
  * ## EXAMPLES
  *
  *    # Copy block by name.
- *    $ wp boilerplate blocks block --name='paragraph'
+ *    $ wp boilerplate blocks use_block --name='paragraph'
```
Comment (dingo-d) [inline src/Cli/ParentGroups/CliBlocks.php:21]: Can we rename these commands to be something like

```bash
wp boilerplate use block --name='paragraph'
```

?

Reads more fluently.

Since we are overhauling all the commands, it makes sense to fix this as well.

File: src/Blocks/UseAssetsCli.php [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+				## RESOURCES
+
+				Assets will be created from this folder:
+				https://github.com/infinum/eightshift-frontend-libs/tree/develop/blocks/init/src/Blocks/assets
+			"),
+		];
+	}
+
+	/* @phpstan-ignore-next-line */
+	public function __invoke(array $args, array $assocArgs) // phpcs:ignore
```
Comment (dingo-d) [inline src/Blocks/UseAssetsCli.php:70]: I was annoyed by this so I released the 1.6.0 version of es coding standards.

You should be able to do the following in the `phpcs.xml.dist`:

```xml
<!-- Add additional extending CLI classes. -->
<rule ref="Eightshift.Commenting.FunctionComment">
    <properties>
        <property name="allowedExtendedClasses" type="array">
            <element value="AbstractCli"/>
            <element value="AbstractBlocksCli"/>
        </property>
    </properties>
</rule>
```

Which should enable you to remove the `// phpcs:ignore` comment from the `__invoke()` magic methods.

Just check all these new CLI methods and what class they are extending and add it to the above list as an `<element>` element in the xml file.

Once you add those, remove the comment and see if it errors out.

File: src/Init/InitProjectCli.php [hunk 13]
```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * Class that registers WPCLI command initial setup of theme project.
```
Comment (dadadavorin) [inline src/Init/InitProjectCli.php:4]:
```suggestion
 * Class that registers WP-CLI command initial setup of theme project.
```

File: src/Init/InitProjectCli.php [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandName(): string
+	{
+		return 'project';
+	}
+
+	/**
+	 * Get WPCLI command doc
```
Comment (dadadavorin) [inline src/Init/InitProjectCli.php:76]:
```suggestion
	 * Get WP-CLI command doc.
```

File: src/Init/InitProjectCli.php [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+							'introOutput' => false,
+						]
+					));
+				}
+			}
+		}
+
+		if (!$groupOutput) {
+			$this->cliLog('--------------------------------------------------');
+			$this->cliLog('We have moved everything you need, to start creating your awesome WordPress project. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
```
Comment (dadadavorin) [inline src/Init/InitProjectCli.php:134]:
```suggestion
			$this->cliLog('We have moved everything you need to start creating your awesome WordPress project. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
```

File: src/Init/InitThemeCli.php [hunk 16]
```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * Class that registers WPCLI command initial setup of theme project.
```
Comment (dadadavorin) [inline src/Init/InitThemeCli.php:4]:
```suggestion
 * Class that registers WP-CLI command initial setup of theme project.
```

File: src/Init/InitThemeCli.php [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+use EightshiftLibs\Menu\MenuCli;
+use ReflectionClass;
+
+/**
+ * Class InitThemeCli
+ */
+class InitThemeCli extends AbstractCli
+{
+	/**
+	 * All classes for initial theme setup for project
```
Comment (dadadavorin) [inline src/Init/InitThemeCli.php:30]:
```suggestion
	 * All classes for initial theme setup for project.
```

File: src/Init/InitThemeCli.php [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+			'type' => 'blocks',
+			'label' => '',
+			'items' => [
+				InitBlocksCli::class,
+			],
+		],
+	];
+
+	/**
+	 * Get WPCLI command parent name
```
Comment (dadadavorin) [inline src/Init/InitThemeCli.php:58]:
```suggestion
	 * Get WP-CLI command parent name.
```

File: src/Init/InitThemeCli.php [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandParentName(): string
+	{
+		return CliInit::COMMAND_NAME;
+	}
+
+	/**
+	 * Get WPCLI command name
```
Comment (dadadavorin) [inline src/Init/InitThemeCli.php:68]:
```suggestion
	 * Get WP-CLI command name.
```

File: src/Init/InitThemeCli.php [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandName(): string
+	{
+		return 'theme';
+	}
+
+	/**
+	 * Get WPCLI command doc
```
Comment (dadadavorin) [inline src/Init/InitThemeCli.php:78]:
```suggestion
	 * Get WP-CLI command doc.
```

File: src/Init/InitThemeCli.php [hunk 21]
```diff
@@ -0,0 +1,140 @@
...
+						]
+					));
+				}
+			}
+
+			$this->cliLog('--------------------------------------------------');
+		}
+
+		if (!$groupOutput) {
+			$this->cliLog('We have moved everything you need, to start creating your awesome WordPress theme. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
```
Comment (dadadavorin) [inline src/Init/InitThemeCli.php:136]:
```suggestion
			$this->cliLog('We have moved everything you need to start creating your awesome WordPress theme. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
```

File: tests/Helpers.php [hunk 22]
```diff
@@ -204,9 +235,72 @@ function deleteCliOutput(string $dir = '') : void
...
+
+	if ($path) {
+		return "{$internalPath}{$sep}{$path}";
+	}
+
+	return $internalPath;
+}
+
+/**
+ * Get projects root path for mocks.
```
Comment (dadadavorin) [inline tests/Helpers.php:268]:
```suggestion
 * Get project root path for mocks.
```

**Review by dadadavorin — COMMENTED** [review]
Wow, amazing work! 🤩 Found few typos/grammar improvements...

File: src/Build/BuildExample.php [hunk 23]
```diff
@@ -3,7 +3,7 @@
 set -e
 
 function build() {
-	cd 'wp-content/themes/eightshift-boilerplate';
+	cd 'wp-content/%project_type%/%project_name%';
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Build/BuildExample.php]: noup removed it

File: src/CiExclude/CiExcludeCli.php (hunk 8 above)
Comment (iruzevic) [reply src/CiExclude/CiExcludeCli.php]: removed

File: src/CiExclude/CiExcludeExample.php [hunk 24]
```diff
@@ -0,0 +1,28 @@
+
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/CiExclude/CiExcludeExample.php]: removed

File: src/Cli/README.md (hunk 9 above)
Comment (iruzevic) [reply src/Cli/README.md:6]: I would leave it if someone comes that they know how to test

File: src/AdminMenus/AdminMenuCli.php [hunk 25]
```diff
@@ -138,7 +113,7 @@ public function getDoc(): array
 				## EXAMPLES
 
 				# Create service class:
-				$ wp boilerplate {$this->getCommandParentName()} {$this->getCommandName()} --title='Content' --menu_title='content' --capability='edit_posts' --menu_slug='es-content'
```
Comment (dingo-d) [inline src/AdminMenus/AdminMenuCli.php:141]: Can you test replacing the `prepareLongDesc` with a heredoc?

https://github.com/infinum/eightshift-libs/blob/fix-install-plugins-method/src/Setup/PluginManageCli.php#L111

File: src/Helpers/Components.php [hunk 26]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+				break;
+			case 'wpContent':
+				$internalPrefix = \dirname(__FILE__, 7);
+
+				if (\getenv('ES_TEST')) {
+					$internalPrefix = \dirname(__FILE__, 3);
+				}
+				break;
+			case 'libs':
+				$path =  self::joinPaths($libsPath);
```
Comment (dingo-d) [inline src/Helpers/Components.php:390]:
```suggestion
				$path = self::joinPaths($libsPath);
```

File: src/Helpers/Components.php [hunk 27]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+
+				if (\getenv('ES_TEST')) {
+					$path = '';
+				}
+				break;
+			case 'blocksGlobalAssetsSource':
+				$path = self::joinPaths([...$flibsPath, $assetsPath]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([...$testsDataPath, $assetsPath]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:400]:
```suggestion
					$path = self::joinPaths([...$testsDataPath, $assetsPath]);
```

File: src/Helpers/Components.php [hunk 28]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+				}
+				break;
+			case 'blocksAssetsSource':
+				$path = self::joinPaths([...$flibsPath, ...$blocksPath, $assetsPath]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([...$testsDataPath, ...$blocksPath, $assetsPath]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:407]:
```suggestion
					$path = self::joinPaths([...$testsDataPath, ...$blocksPath, $assetsPath]);
```

File: src/Helpers/Components.php [hunk 29]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+				}
+				break;
+			case 'blocksStorybookSource':
+				$path = self::joinPaths([...$flibsPath, $storybookPath]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([...$testsDataPath, $storybookPath]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:414]:
```suggestion
					$path = self::joinPaths([...$testsDataPath, $storybookPath]);
```

File: src/Helpers/Components.php [hunk 30]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+				}
+				break;
+			case 'blocksSource':
+				$path = self::joinPaths([...$flibsPath, ...$blocksPath]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([...$testsDataPath, ...$blocksPath]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:421]:
```suggestion
					$path = self::joinPaths([...$testsDataPath, ...$blocksPath]);
```

File: src/Helpers/Components.php [hunk 31]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+				break;
+			case 'blocksSourceWrapper':
+				$name = 'wrapper';
+				break;
+
+			case 'blocksGlobalAssetsDestination':
+				$path = self::joinPaths([$assetsPath]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([$cliOutputPath, $assetsPath]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:441]:
```suggestion
					$path = self::joinPaths([$cliOutputPath, $assetsPath]);
```

File: src/Helpers/Components.php [hunk 32]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+				}
+				break;
+			case 'blocksAssetsDestination':
+				$path = self::joinPaths([...$blocksPath, $assetsPath]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([$cliOutputPath, ...$blocksPath, $assetsPath]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:448]:
```suggestion
					$path = self::joinPaths([$cliOutputPath, ...$blocksPath, $assetsPath]);
```

File: src/Helpers/Components.php [hunk 33]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+				}
+				break;
+			case 'blocksStorybookDestination':
+				$path = self::joinPaths([".{$storybookPath}"]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([$cliOutputPath, ".{$storybookPath}"]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:455]:
```suggestion
					$path = self::joinPaths([$cliOutputPath, ".{$storybookPath}"]);
```

File: src/Helpers/Components.php [hunk 34]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+
+		switch ($type) {
+			case 'blocksSourceCustom':
+			case 'blocksSourceComponents':
+			case 'blocksSourceVariations':
+			case 'blocksSourceWrapper':
+				$path = self::joinPaths([...$flibsPath, ...$blocksPath, $name]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([...$testsDataPath, ...$blocksPath, $name]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:487]:
```suggestion
					$path = self::joinPaths([...$testsDataPath, ...$blocksPath, $name]);
```

File: src/Helpers/Components.php [hunk 35]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+				break;
+
+			case 'blocksDestinationCustom':
+			case 'blocksDestinationComponents':
+			case 'blocksDestinationVariations':
+			case 'blocksDestinationWrapper':
+				$path = self::joinPaths([...$blocksPath, $name]);
+
+				if (\getenv('ES_TEST')) {
+					$path =  self::joinPaths([$cliOutputPath, ...$blocksPath, $name]);
```
Comment (dingo-d) [inline src/Helpers/Components.php:498]:
```suggestion
					$path = self::joinPaths([$cliOutputPath, ...$blocksPath, $name]);
```

File: tests/Helpers/AttributesTraitTest.php [hunk 36]
```diff
@@ -4,37 +4,31 @@
...
-});
-
-afterAll(function() {
-	Monkey\tearDown();
-});
+beforeEach(function () {
+	setBeforeEach();
 
-beforeEach(function() {
+	buildTestBlocks();
```
Comment (dingo-d) [inline tests/Helpers/AttributesTraitTest.php:18]: Try to see if you can use `uses` to bind this to just this test case.

https://pestphp.com/docs/underlying-test-case#uses

**Review by dingo-d — CHANGES_REQUESTED** [review]
Left a few comments to fix. Nothing major imo.

Can we change all the command arguments so that they are using `-` instead of `_` as a separator?
That's the way the default CLI commands use.

File: src/Blocks/BlocksCli.php [hunk 37]
```diff
@@ -94,7 +36,7 @@ public function getCommandParentName(): string
 	 */
 	public function getCommandName(): string
 	{
-		return 'blocks';
+		return 'create blocks_class';
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Blocks/BlocksCli.php]: done

File: src/AdminMenus/AdminMenuCli.php (hunk 25 above)
Comment (dingo-d) [reply src/AdminMenus/AdminMenuCli.php:141]: Oh, I see it now! Wasn't aware that the headings will be parsed that way 👍🏼

File: tests/Init/InitBlocksCliTest.php [hunk 38]
```diff
@@ -0,0 +1,47 @@
...
+	setAfterEach();
+
+	unset($this->mock);
+});
+
+test('Blocks CLI command will correctly copy the Blocks class with defaults', function () {
+	$mock = $this->mock;
+	$mock([], []);
+
+	// $output = \file_get_contents(Components::getProjectPaths('blocksDestination', "Blocks.php"));
```
Comment (iobrado) [inline tests/Init/InitBlocksCliTest.php:26]: Do you need this commented out part or can it be removed?

**Review by iobrado — CHANGES_REQUESTED** [review]
Most impressive PR 👏  I left suggestions to fix some typos I stumbled upon.

File: tests/Init/InitBlocksCliTest.php (hunk 38 above)
Comment (iruzevic) [reply tests/Init/InitBlocksCliTest.php:26]: fixed

File: src/Blocks/BlocksCli.php [hunk 39]
```diff
@@ -106,92 +48,37 @@ public function getDoc(): array
 	{
 		return [
 			'shortdesc' => 'Create blocks service class.',
-			'longdesc' => "
+			'longdesc' => $this->prepareLongDesc("
+				## USAGE
+
 				This file is a main entrypoint for all our block editor setup.
```
Comment (dadadavorin) [inline src/Blocks/BlocksCli.php:54]:
```suggestion
				This file is a main entry point for all our block editor setup.
```

File: src/Blocks/UseAssetsCli.php [hunk 40]
```diff
@@ -0,0 +1,94 @@
...
+use EightshiftLibs\Helpers\Components;
+use WP_CLI;
+
+/**
+ * Class UseAssetsCli
+ */
+class UseAssetsCli extends AbstractBlocksCli
+{
+	/**
+	 * Get WPCLI command parent name
```
Comment (dadadavorin) [inline src/Blocks/UseAssetsCli.php:23]:
```suggestion
	 * Get WP-CLI command parent name
```

File: src/Blocks/UseAssetsCli.php [hunk 41]
```diff
@@ -0,0 +1,94 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandParentName(): string
+	{
+		return CliBlocks::COMMAND_NAME;
+	}
+
+	/**
+	 * Get WPCLI command name
```
Comment (dadadavorin) [inline src/Blocks/UseAssetsCli.php:33]:
```suggestion
	 * Get WP-CLI command name
```

File: src/Blocks/UseAssetsCli.php [hunk 42]
```diff
@@ -0,0 +1,94 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandName(): string
+	{
+		return 'use-assets';
+	}
+
+	/**
+	 * Get WPCLI command doc
```
Comment (dadadavorin) [inline src/Blocks/UseAssetsCli.php:43]:
```suggestion
	 * Get WP-CLI command doc
```

File: src/Blocks/UseGlobalAssetsCli.php [hunk 43]
```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Class that registers WPCLI command for Blocks global assets.
```
Comment (dadadavorin) [inline src/Blocks/UseGlobalAssetsCli.php:4]:
```suggestion
 * Class that registers WP-CLI command for Blocks global assets.
```

File: src/Blocks/UseGlobalAssetsCli.php [hunk 44]
```diff
@@ -0,0 +1,95 @@
...
+use EightshiftLibs\Helpers\Components;
+use WP_CLI;
+
+/**
+ * Class UseGlobalAssetsCli
+ */
+class UseGlobalAssetsCli extends AbstractBlocksCli
+{
+	/**
+	 * Get WPCLI command parent name
```
Comment (dadadavorin) [inline src/Blocks/UseGlobalAssetsCli.php:23]:
```suggestion
	 * Get WP-CLI command parent name
```

File: src/Blocks/UseGlobalAssetsCli.php [hunk 45]
```diff
@@ -0,0 +1,95 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandParentName(): string
+	{
+		return CliBlocks::COMMAND_NAME;
+	}
+
+	/**
+	 * Get WPCLI command name
```
Comment (dadadavorin) [inline src/Blocks/UseGlobalAssetsCli.php:33]:
```suggestion
	 * Get WP-CLI command name
```

File: src/Blocks/UseGlobalAssetsCli.php [hunk 46]
```diff
@@ -0,0 +1,95 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandName(): string
+	{
+		return 'use-global-assets';
+	}
+
+	/**
+	 * Get WPCLI command doc
```
Comment (dadadavorin) [inline src/Blocks/UseGlobalAssetsCli.php:43]:
```suggestion
	 * Get WP-CLI command doc
```

File: src/Blocks/UseManifestCli.php [hunk 47]
```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * Class that registers WPCLI command for Blocks Manifest.
```
Comment (dadadavorin) [inline src/Blocks/UseManifestCli.php:4]:
```suggestion
 * Class that registers WP-CLI command for Blocks Manifest.
```

File: src/Blocks/UseManifestCli.php [hunk 48]
```diff
@@ -0,0 +1,93 @@
...
+use EightshiftLibs\Helpers\Components;
+use WP_CLI;
+
+/**
+ * Class UseManifestCli
+ */
+class UseManifestCli extends AbstractBlocksCli
+{
+	/**
+	 * Get WPCLI command parent name
```
Comment (dadadavorin) [inline src/Blocks/UseManifestCli.php:23]:
```suggestion
	 * Get WP-CLI command parent name
```

File: src/Blocks/UseWrapperCli.php [hunk 49]
```diff
@@ -0,0 +1,91 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandName(): string
+	{
+		return 'use-wrapper';
+	}
+
+	/**
+	 * Get WPCLI command doc
```
Comment (dadadavorin) [inline src/Blocks/UseWrapperCli.php:43]:
```suggestion
	 * Get WP-CLI command doc
```

File: src/Init/InitAllCli.php [hunk 50]
```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * Class that registers WPCLI command initial setup for all.
```
Comment (dadadavorin) [inline src/Init/InitAllCli.php:4]:
```suggestion
 * Class that registers WP-CLI command initial setup for all.
```

File: src/Init/InitAllCli.php [hunk 51]
```diff
@@ -0,0 +1,130 @@
...
+			'type' => 'blocks',
+			'label' => '',
+			'items' => [
+				InitBlocksCli::class,
+			],
+		],
+	];
+
+	/**
+	 * Get WPCLI command parent name.
```
Comment (dadadavorin) [inline src/Init/InitAllCli.php:44]:
```suggestion
	 * Get WP-CLI command parent name.
```

File: src/Init/InitAllCli.php [hunk 52]
```diff
@@ -0,0 +1,130 @@
...
+	 *
+	 * @return string
+	 */
+	public function getCommandParentName(): string
+	{
+		return CliInit::COMMAND_NAME;
+	}
+
+	/**
+	 * Get WPCLI command name.
```
Comment (dadadavorin) [inline src/Init/InitAllCli.php:54]:
```suggestion
	 * Get WP-CLI command name.
```

**Review by dadadavorin — DISMISSED** [review]
Looks good 👍  Just found a few typos..

File: src/Blocks/AbstractBlocksCli.php [hunk 53]
```diff
@@ -19,152 +20,205 @@
...
-		// Set optional arguments.
-		$skipExisting = $this->getSkipExisting($assocArgs);
+		// Clean up name.
+		$name = $args['name'] ?? '';
+		$name = \str_replace(' ', '', $name);
+		$name = \trim($name, \DIRECTORY_SEPARATOR);
 
-		$root = $this->getProjectRootPath();
-		$rootNode = $this->getFrontendLibsBlockPath();
+		$isFile = \strpos($name, '.') !== false;
```
Comment (mbmjertan) [inline src/Blocks/AbstractBlocksCli.php:45]: This is not a valid assumption to have on UNIX systems, e.g. `/etc/hosts` is a file. All files that'll be moved probably have an extension, but this is not a valid way to check whether something is a file.

Rather use [`is_file`](https://www.php.net/manual/en/function.is-file.php) and pass the file path to it.

File: src/Blocks/BlocksCli.php (hunk 39 above)
Comment (mbmjertan) [reply src/Blocks/BlocksCli.php:54]: "for all our block editor setup" also doesn't tell much, maybe rephrase that part as well

File: src/Helpers/Components.php [hunk 54]
```diff
@@ -296,18 +275,252 @@ public static function getManifest(string $path, string $name = ''): array
...
+			$prefix = $internalPrefix;
+		}
+
+		$path = self::joinPaths([$prefix, $path, $sufix]);
+
+		return \str_replace("{$sep}{$sep}", $sep, $path);
+	}
+
+	/**
+	 * Paths join
```
[the remark this answers is not in the document]
Comment (mbmjertan) [reply src/Helpers/Components.php:513]:
```suggestion
	 * Join paths.
```

#wordOrderMatters

File: src/Init/InitAllCli.php [hunk 55]
```diff
@@ -0,0 +1,130 @@
...
+						$assocArgs,
+						[
+							'groupOutput' => $type === 'blocks',
+							'introOutput' => false,
+						]
+					));
+				}
+			}
+
+			$this->cliLog('--------------------------------------------------');
```
Comment (mbmjertan) [inline src/Init/InitAllCli.php:122]: Do you copy/paste these separators? Maybe we should have a helper for that as well, so it's consistent.

File: src/Init/InitBlocksCli.php [hunk 56]
```diff
@@ -0,0 +1,232 @@
...
+				}
+			}
+		}
+
+		$this->getInitBlocks($assocArgs, $commands);
+
+		if (!$groupOutput) {
+			$this->cliLog('--------------------------------------------------');
+			$this->cliLog('We have moved everything you need to start creating WordPress blocks. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
+			$this->cliLog('Happy developing!', "M");
```
Comment (mbmjertan) [inline src/Init/InitBlocksCli.php:186]: As previously

File: src/Init/InitProjectCli.php [hunk 57]
```diff
@@ -0,0 +1,134 @@
...
+					));
+				}
+			}
+		}
+
+		if (!$groupOutput) {
+			$this->cliLog('--------------------------------------------------');
+			$this->cliLog('We have moved everything you need to start creating your awesome WordPress project. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
+			$this->cliLog('Happy developing!', "M");
+		}
```
Comment (mbmjertan) [inline src/Init/InitProjectCli.php:132]: As above

File: src/Init/InitThemeCli.php [hunk 58]
```diff
@@ -0,0 +1,140 @@
...
+
+			$this->cliLog('--------------------------------------------------');
+		}
+
+		if (!$groupOutput) {
+			$this->cliLog('We have moved everything you need to start creating your awesome WordPress theme. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
+			$this->cliLog('Happy developing!', "M");
+		}
+	}
+}
```
Comment (mbmjertan) [inline src/Init/InitThemeCli.php:140]: As above

**Review by mbmjertan — CHANGES_REQUESTED** [review]
Reviewed files changed, apart from tests, as it's been quite a task to review this. Congratulations, this was an amazing endeavor. I've left some comments about phrasing, and some tips for a potentially better approach in some places, but I don't take issue with this code. Great work.

File: src/Init/InitAllCli.php [hunk 59]
```diff
@@ -0,0 +1,130 @@
...
+		}
+
+		if (!$groupOutput) {
+			$this->cliLog('We have moved everything we have to your project. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
+			$this->cliLog('Happy developing!', "M");
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Init/InitAllCli.php:127]: I have created an issue for this https://github.com/infinum/eightshift-libs/issues/310
